<img src="imgs/unifor_logo.png" width="400">
<br>
<b>
<font size="6" face="arial" color="blue">
    Graduação em Ciência da Computação
</font>
</b>
<br
<b>
<font size="4" face="arial">
    Disciplina: Métodos Quantitativos para Computação
</font>
</b>

**Orientador: Prof. Me. Ricardo Carubbi** <br>
*Docente da Graduação e Pós-Graduação em Ciência de Dados e Inteligência Artificial*<br>
*Laboratório de Ciência de Dados e Inteligência Artificial*<br>
*Universidade de Fortaleza*<br>

<p dir="ltr" style="text-align: left;">
    <strong>Links:</strong>
    <a href="https://www2.unifor.br/controle_pesquisa/pesquisarprofessor.do?actionParameter=prepareUpdate&amp;p_tp_ambiente=2&amp;p_tp_chamada=1&amp;p_tp_apresentacao=1&amp;cdPesquisador=767686193" target="_blank">Unifor</a|
    <a href="http://lattes.cnpq.br/5738786447903616" target="_blank">Lattes</a|
    <a href="https://unifor.br/web/pesquisa-inovacao/ncdia" target="_blank">NCDIA</a|
    <a href="https://github.com/carubbi/" target="_blank">Github</a>
</p>

# **Testes de Hipóteses**

Um problema muito comum enfrentado por cientistas é a **avaliação de significância** em dados estatísticos dispersos. Devido à disponibilidade limitada de dados observacionais, os cientistas aplicam métodos estatísticos inferenciais para decidir se os dados observados contêm informações significativas ou se os dados dispersos não passam de uma manifestação da natureza inerentemente probabilística do processo de geração de dados.

De modo geral, um cientista formula tal problema da seguinte maneira: ele constrói um modelo que simplifica o processo de geração de dados e considera uma suposição particular – uma chamada **hipótese** – desse modelo. Com base nos dados, ele deseja avaliar essa hipótese.

A metodologia dos testes de hipóteses trata de fazer **inferências estatísticas sobre populações com base em amostras retiradas da população**. Uma maneira de estimar um parâmetro populacional é a construção de intervalos de confiança. Outra maneira é tomar uma decisão sobre um parâmetro na forma de um teste. Qualquer teste de hipótese envolve a coleta de dados (amostragem). Se a hipótese for considerada correta, o cientista pode calcular os resultados esperados de um experimento. Se os dados observados diferem significativamente dos resultados esperados, considera-se que a hipótese ou suposição está incorreta. Assim, com base nos dados observados, o cientista decide se há evidências suficientes, a partir da análise dos dados, de que o modelo – a hipótese – deve ser rejeitado ou se há evidências insuficientes para rejeitar a hipótese declarada.

## **Introdução aos Testes de Hipóteses**

A **estatística inferencial** trata da tomada de decisões ou julgamentos sobre o valor de uma observação ou medição específica. Um dos métodos mais comumente usados para tomar tais decisões é realizar um **teste de hipótese**. Uma **hipótese** é uma explicação proposta para um fenômeno. No contexto dos testes de hipóteses estatísticos, o termo hipótese é uma declaração sobre algo que se supõe ser verdadeiro.

Um teste de hipótese envolve duas hipóteses: a **hipótese nula** e a **hipótese alternativa**. A hipótese nula ($H_0$) é a declaração a ser testada. A hipótese alternativa ($H_A$ ou $H_1$) é uma declaração considerada como uma alternativa à hipótese nula.

**O teste de hipótese tem como objetivo testar se a hipótese nula deve ser rejeitada em favor da hipótese alternativa**. A lógica básica de um teste de hipótese é comparar dois conjuntos de dados estatísticos. Um conjunto de dados é obtido por amostragem, e o outro se origina de um modelo idealizado. A hipótese nula **não é rejeitada** se os dados da amostra forem consistentes com o modelo idealizado. Se os dados da amostra forem inconsistentes com o modelo idealizado e, assim, apoiarem uma hipótese alternativa, a hipótese nula é **rejeitada** em favor da hipótese alternativa.

O critério para rejeitar a hipótese nula envolve uma chamada de uma **estatística do teste**. A estatística do teste é um número calculado a partir do conjunto de dados, obtido por medições e observações, ou, de forma mais geral, por amostragem.

### Vídeo: O que é Teste de Hipóteses e a Hipótese Nula??

Assista ao vídeo explicativo sobre **Teste de Hipóteses e a Hipótese Nula** neste [link](https://youtu.be/0oc49DyA3hU?si=QcXxXa_RRzotAuLz):

<img src="https://drive.google.com/uc?id=1NAdpfPDWPpaqx8jlDfL0ifrxhNFJPOxl" width="400">

### Vídeo: O que é Hipótese Alternativa??

Assista ao vídeo explicativo sobre **Hipótese Alternativa** neste [link](https://youtu.be/5koKb5B_YWo?si=9AzizJjNFDm3HFFD):

<img src="https://drive.google.com/uc?id=1pBb9ea5xAy-hds0bGBA2RcoO39r_op0V" width="400">

## **Formulação das Hipóteses**

Qualquer teste de hipótese começa com a formulação da **hipótese nula** e da **hipótese alternativa**. Esta seção foca em testes de hipóteses para uma média populacional, $\mu$. No entanto, o procedimento geral se aplica a qualquer teste de hipótese.

A hipótese nula para um teste de hipótese referente a uma média populacional, $\mu$, é expressa como:
$$H_0: \mu = \mu_0,$$
onde $\mu_0$ é algum número (o valor de referência a ser testado).

A formulação da hipótese alternativa depende do objetivo do teste de hipótese. Existem três maneiras de formular uma hipótese alternativa (Weiss, 2010):

1.  Se o teste de hipótese visa decidir se uma média populacional, $\mu$, é **diferente** do valor especificado $\mu_0$, a hipótese alternativa é expressa como:
$$H_A: \mu \ne \mu_0.$$
Tal teste de hipótese é chamado de **teste bilateral** (ou bicaudal).

2.  Se o teste de hipótese visa decidir se uma média populacional, $\mu$, é **menor** que o valor especificado $\mu_0$, a hipótese alternativa é expressa como:
$$H_A: \mu < \mu_0.$$
Tal teste de hipótese é chamado de **teste unilateral esquerdo** (ou de cauda esquerda).

3.  Se o teste de hipótese visa decidir se uma média populacional, $\mu$, é **maior** que o valor especificado $\mu_0$, a hipótese alternativa é expressa como:
$$H_A: \mu > \mu_0.$$
Tal teste de hipótese é chamado de **teste unilateral direito** (ou de cauda direita).

> **Nota:** Um teste de hipótese é chamado de **teste unilateral** se ele for unilateral esquerdo ou unilateral direito.

| | Teste Bilateral | Teste Unilateral Esquerdo | Teste Unilateral Direito |
| :--- | :---: | :---: | :---: |
| **Sinal em $H_A$** | $\ne$ | $<$ | $>$ |
| **Região de Rejeição** | Ambos os lados | Lado esquerdo | Lado direito |

## **Erro Tipo I, Erro Tipo II e Nível de Significância**

Qualquer decisão tomada com base em um teste de hipótese pode estar incorreta. Na metodologia dos testes de hipóteses, existem dois tipos de erros: **Erro Tipo I** e **Erro Tipo II**.

Um **erro tipo I** ocorre quando uma hipótese nula verdadeira é rejeitada (um **“falso positivo”**), enquanto um **erro tipo II** ocorre quando uma hipótese nula falsa não é rejeitada (um **“falso negativo”**). Em outras palavras, um erro tipo I detecta um efeito que não está presente, enquanto um erro tipo II falha em detectar um efeito que está presente.

| Decisão | $H_0$ é verdadeira | $H_0$ é falsa |
| :--- | :---: | :---: |
| **Não rejeitar $H_0$** | Decisão Correta | Erro Tipo II ($\beta$) |
| **Rejeitar $H_0$** | Erro Tipo I ($\alpha$) | Decisão Correta (Poder do Teste, $1-\beta$) |

Se você está confuso sobre os erros tipo I e tipo II, pode achar esta ilustração útil ([aqui](https://effectsizefaq.com/2010/05/31/i-always-get-confused-about-type-i-and-ii-errors-can-you-show-me-something-to-help-me-remember-the-difference/)).

Conduzir um teste de hipótese sempre implica que há uma chance de tomar uma decisão incorreta. A probabilidade do erro tipo I (uma hipótese nula verdadeira ser rejeitada) é comumente chamada de **nível de significância** do teste de hipótese e é denotada por **$\alpha$**. A probabilidade de um erro tipo II (uma hipótese nula falsa não ser rejeitada) é denotada por **$\beta$**. Lembre-se que, para um tamanho de amostra fixo, quanto menor especificarmos o nível de significância, $\alpha$, maior será a probabilidade, $\beta$, de não rejeitar uma hipótese nula falsa.

O resultado de um teste de hipótese é uma declaração a favor da hipótese nula ou a favor da hipótese alternativa. Se a hipótese nula é rejeitada, os dados fornecem evidências suficientes para apoiar a hipótese alternativa. Se a hipótese nula não é rejeitada, os dados não fornecem evidências suficientes para apoiar a hipótese alternativa. Se o teste de hipótese é realizado no nível de significância $\alpha$ e a hipótese nula é rejeitada, pode-se afirmar que os resultados do teste são **estatisticamente significantes** no nível $\alpha$. Se a hipótese nula não é rejeitada no nível de significância $\alpha$, pode-se afirmar que os resultados do teste **não são estatisticamente significantes** no nível $\alpha$.

## **A Abordagem do Valor Crítico e a Abordagem do Valor-p para Testes de Hipóteses**

Para decidir se a hipótese nula deve ser rejeitada, uma **estatística do teste** é calculada. A decisão é tomada com base no valor numérico dessa estatística do teste. Existem duas abordagens para chegar a essa decisão:

1.  A **abordagem do valor crítico**.
2.  A **abordagem do valor-p** (p-value).

### **A Abordagem do Valor Crítico**

Ao aplicar a **abordagem do valor crítico**, determina-se se a estatística do teste observada é ou não mais extremo que um valor crítico definido. Portanto, a estatística do teste observada (calculada com base nos dados da amostra) é comparada ao valor crítico (uma espécie de valor de corte, ou *cutoff*). A hipótese nula é rejeitada se a estatística do teste for mais extrema que o valor crítico.

A hipótese nula não é rejeitada se a estatística do teste não for tão extrema quanto o valor crítico. O valor crítico é calculado com base no nível de significância $\alpha$ fornecido e no tipo de distribuição de probabilidade do modelo idealizado. O valor crítico divide a área sob a curva de distribuição de probabilidade em **região(ões) de rejeição** e **região de não rejeição**.

As três figuras a seguir mostram um teste bilateral, unilateral direito e unilateral esquerdo. O modelo idealizado nas figuras, e portanto $H_0$, é descrito por uma curva de distribuição normal em forma de sino.

* Em um **teste bilateral**, a hipótese nula é rejeitada se a estatística do teste for muito pequeno ou muito grande. Assim, a região de rejeição para tal teste consiste em duas partes: uma à esquerda e outra à direita.

<img src="https://drive.google.com/uc?id=1mj6NP_9Wlek9GPT4V7gn1qgEbeoxKwfH" width="600">


* A hipótese nula é rejeitada em um **teste unilateral esquerdo** se o estatística do teste for muito pequen0. Assim, a região de rejeição para tal teste consiste em uma parte à esquerda do centro.

<img src="https://drive.google.com/uc?id=13hSu2ZlGqHCa3JcGbYVvseC5avY5Aorp" width="600">

* A hipótese nula é rejeitada em um **teste unilateral direito** se o estatística do teste for muito grande. Assim, a região de rejeição para tal teste consiste em uma parte à direita do centro.

<img src="https://drive.google.com/uc?id=1X8Whp-xR230lUcsR0oC5N2T7ubwFsMl4" width="600">

### **A Abordagem do Valor-p**

Para a **abordagem do valor-p** (p-value), a probabilidade (o *valor-p*) do valor numérico do estatística do teste é comparada ao nível de significância ($\alpha$) especificado para o teste.

O **valor-p** corresponde à probabilidade de observar dados amostrais *pelo menos tão extremos* quanto o estatística do teste realmente obtida. Valores-p pequenos fornecem evidência contra a hipótese nula. Quanto menor (mais próximo de 0) o valor-p, mais forte é a evidência contra a hipótese nula.

A hipótese nula é rejeitada se o **valor-p for menor ou igual** ao nível de significância $\alpha$ especificado. Caso contrário, a hipótese nula não é rejeitada.

> **Nota (Regra de Decisão):** Se **$p \le \alpha$**, rejeite $H_0$; caso contrário, se **$p > \alpha$**, não rejeite $H_0$.

#### Vídeo: O que são Valores-p: O que são e como interpretá-los??

Assista ao vídeo explicativo sobre **Valores-p: O que são e como interpretá-los** neste [link](https://youtu.be/vemZtEM63GY?si=QQwZBJZK-dBP1UuY):

<img src="https://drive.google.com/uc?id=1CuADHi8u2jKcT_OgBwxZXxLKKozQeAk8" width="400">

Consequentemente, ao conhecer o valor-p, qualquer nível de significância desejado pode ser avaliado. Por exemplo, se o valor-p de um teste de hipótese é 0.01, a hipótese nula pode ser rejeitada em qualquer nível de significância maior ou igual a 0.01 (ex: 0.05 ou 0.10). Ela não é rejeitada em qualquer nível de significância menor que 0.01 (ex: 0.005). Assim, o valor-p é comumente usado para avaliar a **força da evidência** contra a hipótese nula, sem referência a um nível de significância pré-definido.

A tabela a seguir fornece diretrizes para usar o valor-p na avaliação da evidência contra a hipótese nula (Weiss, 2011):

| Valor-p | Evidência contra $H_0$            |
| ---     | ---                               |
| $$p > 0.10$$ | Evidência fraca ou inexistente |
| $$0.05 < p \le 0.10$$ | Evidência moderada    |
| $$0.01 < p \le 0.05$$ | Evidência forte       |
| $$p \le 0.01$$ | Evidência muito forte        |

## **Testes de Hipótese para uma Média Populacional**

A forma mais simples de um teste de hipótese é o teste para uma média populacional. Neste teste, comparamos a média, $\mu$, obtida da amostragem (observações) com uma suposta média populacional, $\mu_0$. A hipótese nula afirma que a média e a média populacional são iguais e é assim escrita como:
$$H_0: \mu = \mu_0$$

Se a média dos dados observados ($\mu$) diferir significativamente da suposta média populacional ($\mu_0$), então a suposição, $H_0$, é rejeitada em favor da hipótese alternativa, $H_A$. Dependendo da questão de pesquisa específica, a hipótese alternativa, $H_A$, é escrita como:
$$H_A: \mu < \mu_0 \quad \text{ou} \quad \mu > \mu_0 \quad \text{ou} \quad \mu \neq \mu_0$$

Se, no entanto, os dados não fornecerem evidências suficientes para rejeitar a hipótese declarada ($H_0$), nós **não rejeitamos $H_0$**. Neste caso, concluímos que os dados não fornecem evidências suficientes para assumir que a média observada ($\mu$) é diferente da suposta média populacional ($\mu_0$). A variabilidade observada nos dados é atribuída à natureza inerentemente probabilística do processo de geração de dados. Em outras palavras, a variabilidade observada nos dados deve-se ao **acaso**.

Nas seções seguintes, mostraremos que o conhecimento que temos sobre os parâmetros da população é relevante para o cálculo real da significância estatística.

# **Testes de Hipótese para uma Média Populacional quando $\sigma$ (Sigma) é Conhecido**

Se o desvio padrão populacional ($\sigma$) é conhecido, um teste de hipótese para uma média populacional é chamado de **teste z para uma média** ou simplesmente **teste z**.

O teste z é um teste de hipótese para testar uma média populacional, $\mu$, contra uma suposta média populacional, $\mu_0$. O teste z pressupõe que as variáveis são normalmente distribuídas ou que o tamanho da amostra é grande; neste último caso, o **Teorema do Limite Central** garante uma **distribuição amostral** normalmente distribuída.

Além disso, o desvio padrão da população, $\sigma$, deve ser conhecido. Em aplicações da vida real, essa suposição quase nunca é atendida. Portanto, o teste z é raramente aplicado. No entanto, ele é o teste de hipótese mais simples e, por isso, um bom tópico para começar.

Para realizar o teste z, seguimos o procedimento passo a passo mostrado abaixo. Primeiro, apresentaremos a **abordagem do valor crítico**. Em segundo lugar, repetiremos a análise para a **abordagem do p-valor**.

1.  Definir a hipótese nula $H_0$ e a hipótese alternativa $H_A$.
2.  Decidir o nível de significância, $\alpha$.
3.  Calcular o valor do **estatística do teste**.
4.  **Abordagem do valor crítico:** Determinar o valor crítico.
5.  **Abordagem do p-valor:** Determinar o p-valor.
6.  **Abordagem do valor crítico:** Se o valor do estatística do teste cair na **região de rejeição**, rejeite $H_0$; caso contrário, **não rejeite $H_0$**.
7.  **Abordagem do p-valor:** Se $p \le \alpha$, rejeite $H_0$; caso contrário, **não rejeite $H_0$**.
8.  Interpretar o resultado do teste de hipótese.

### **Teste z para uma média: Um exemplo**

Nesta seção, trabalharemos novamente com o conjunto de dados `students`. Você pode baixar o arquivo `students.csv` [aqui](https://www.google.com/search?q=PLACEHOLDER_FOR_LINK) e importá-lo do seu sistema de arquivos local, ou carregá-lo diretamente como um recurso da web. Em ambos os casos, você importa o conjunto de dados para o Python como um objeto DataFrame do `pandas` usando o método `read_csv`:

In [69]:
import numpy as np
import pandas as pd

In [70]:
link = "https://drive.google.com/file/d/1PCnR6WzmtIvvoHWoNo8-aZ4lmuoQNM5i/view?usp=drive_link"
file_id = link.split("/")[5]
download_url = f'https://drive.google.com/uc?export=download&id={file_id}'
students = pd.read_csv(download_url)

students.head(10)

,stud.id,name,gender,age,height,weight,religion,nc.score,semester,major,minor,score1,score2,online.tutorial,graduated,salary
1,833917,"Gonzales, Christina",Female,19,160,64.8,Muslim,1.91,1st,Political Science,Social Sciences,NaN,NaN,0,0,NaN
2,898539,"Lozano, T'Hani",Female,19,172,73.0,Other,1.56,2nd,Social Sciences,Mathematics and Statistics,NaN,NaN,0,0,NaN
3,379678,"Williams, Hanh",Female,22,168,70.6,Protestant,1.24,3rd,Social Sciences,Mathematics and Statistics,45.0,46.0,0,0,NaN
4,807564,"Nem, Denzel",Male,19,183,79.7,Other,1.37,2nd,Environmental Sciences,Mathematics and Statistics,NaN,NaN,0,0,NaN
5,383291,"Powell, Heather",Female,21,175,71.4,Catholic,1.46,1st,Environmental Sciences,Mathematics and Statistics,NaN,NaN,0,0,NaN
6,256074,"Perez, Jadrian",Male,19,189,85.8,Catholic,1.34,2nd,Political Science,Mathematics and Statistics,NaN,NaN,0,0,NaN
7,754591,"Clardy, Anita",Female,21,156,65.9,Protestant,1.11,2nd,Political Science,Social Sciences,NaN,NaN,0,0,NaN
8,146494,"Allen, Rebecca Marie",Female,21,167,65.7,Other,2.03,3rd,Political Science,Economics and Finance,58.0,62.0,0,0,NaN
9,723584,"Tracy, Robert",Male,18,195,94.4,Other,1.29,3rd,Economics and Finance,Environmental Sciences,57.0,67.0,0,0,NaN
10,314281,"Nimmons, Laura",Female,18,165,66.0,Orthodox,1.19,2nd,Environmental Sciences,Mathematics and Statistics,NaN,NaN,0,0,NaN


O conjunto de dados `students` consiste em 8239 linhas, cada uma representando um estudante específico, e 16 colunas, cada uma correspondendo a uma variável/feature (característica) relacionada àquele estudante. Estas variáveis, cujos nomes são autoexplicativos, são:

* `stud.id`
* `name`
* `gender`
* `age`
* `height`
* `weight`
* `religion`
* `nc.score`
* `semester`
* `major`
* `minor`
* `score1`
* `score2`
* `online.tutorial`
* `graduated`
* `salary`

Para exemplificar o teste de hipótese, examinamos o `weight` (peso) médio dos estudantes e o comparamos com o `weight` (peso) médio de adultos europeus. Walpole et al. (2012) publicaram dados sobre a massa corporal média (kg) por região, incluindo a Europa. Eles relatam que a massa corporal média da população adulta europeia é de **70.8 kg**.

Nós definimos a média populacional (de referência) adequadamente: $\mu_0 = 70.8$.

Infelizmente, devido à abordagem metodológica de Walpole et al. (2012), eles não forneceram um **desvio padrão ($\sigma$)** dos pesos dos adultos europeus.

**Para fins de demonstração**, vamos *assumir* que os dados de peso (`weight`) fornecidos no conjunto de dados `students` são uma boa aproximação para a população de interesse. Assim, definimos $\sigma$ como o desvio padrão $s$ da variável `weight` no conjunto de dados `students`:

In [71]:
mu_0 = 70.8
sigma = np.std(students["weight"])

print(round(sigma, 2))

8.63


Além disso, pegamos uma amostra aleatória com tamanho $n = 14$. A amostra consiste nos `weights` (pesos) em `kg` de 14 estudantes escolhidos aleatoriamente do conjunto de dados `students`. Finalmente, calculamos a média amostral ($\bar{x}$), nossa estatística amostral de interesse. A estatística amostral é atribuída à variável `mean_sample`.

In [72]:
n = 14
sample_weights = students.sample(n, random_state = 8)["weight"]

mean_sample = np.mean(sample_weights)

print(round(mean_sample, 2))

69.82


### **Teste de hipótese: A abordagem do valor crítico**

**Passo 1: Definir a hipótese nula, $H_0$, e a hipótese alternativa, $H_A$.**

A hipótese nula afirma que o peso médio dos estudantes ($\mu$) é igual ao peso médio dos adultos europeus de 70.8 kg ($\mu_0$), conforme relatado por Walpole et al. (2012). Em outras palavras, não há diferença no peso médio dos estudantes e no peso médio dos adultos europeus.
$$H_0: \mu = 70.8$$

Para fins de ilustração, testaremos três hipóteses alternativas:

**Hipótese alternativa 1:** O peso médio dos estudantes **não é igual** ao peso médio dos adultos europeus. Em outras palavras, *existe* uma diferença no peso médio dos estudantes e no peso médio dos adultos europeus.
$$H_{A1}: \mu \neq 70.8$$

**Hipótese alternativa 2:** O peso médio dos estudantes é **menor que** o peso médio dos adultos europeus.
$$H_{A2}: \mu < 70.8$$

**Hipótese alternativa 3:** O peso médio dos estudantes é **maior que** o peso médio dos adultos europeus.
$$H_{A3}: \mu > 70.8$$


**Passo 2: Decidir o nível de significância, $\alpha$.**
$$\alpha = 0.05$$

In [73]:
alpha = 0.05

**Passo 3: Calcular o valor do estatística do teste.**

A seguinte equação é aplicada para calcular o estatística do teste $z$:

$$z = \frac{\bar{x} - \mu_0}{\sigma / \sqrt{n}}$$

Aplicado ao nosso exemplo, calculamos $z$ por:

$$z = \frac{\bar{x} - \mu_0}{\sigma / \sqrt{n}} = \frac{69.82 - 70.8}{8.64 / \sqrt{14}} \approx -0.42$$

In [ ]:
z = (mean_sample - mu_0) / (sigma / np.sqrt(n))

print(z)

-0.4240454748453689


**Passo 4A: Determinar o valor crítico.**

Para calcular o **valor crítico**, usamos a função `norm.ppf()` derivada do pacote `scipy`. Lembre-se, estamos testando três hipóteses alternativas ($H_{A1}$, $H_{A2}$ e $H_{A3}$). Portanto, temos que calcular três valores críticos também ($z_{A1} = \pm z_{\alpha/2}$, $z_{A2} = -z_{\alpha}$ e $z_{A3} = +z_{\alpha}$).

In [ ]:
from scipy import stats
z_ha1 = stats.norm.ppf(alpha / 2)
z_ha2 = stats.norm.ppf(alpha)
z_ha3 = stats.norm.ppf(1 - alpha)
print(z_ha1, z_ha2, z_ha3)

-1.9599639845400545 -1.6448536269514729 1.6448536269514722


Os valores críticos são $z_{A1} \approx \pm 1.96$, $z_{A2} \approx -1.64$ e $z_{A3} \approx 1.64$

**Passo 5a: Se o valor do estatística do teste cair na região de rejeição, rejeite $H_0$; caso contrário, não rejeite $H_0$.**

O valor do estatística do teste encontrado no Passo 3 é $z \approx -0.42$.

Lembre-se, estamos investigando três hipóteses alternativas ($H_{A1}$, $H_{A2}$ e $H_{A3}$). Assim, avaliamos a região de rejeição para cada hipótese específica.

**$H_{A1}: \mu \neq 70.8$**

Lembre-se dos valores críticos para $H_{A1}$:
$$z_{A1} = \pm z_{\alpha/2} = \pm 1.96$$

O estatística do teste ($z \approx -0.42$) cai na região de rejeição? Esteja ciente de que o teste é **bicaudal** (two-tailed), o que significa que avaliamos o limite superior e o inferior.

Comparando o estatística do teste com o limite superior:
$$-0.42 > 1.96$$

In [74]:
print(z > np.abs(z_ha1))

False


Comparando o estatística do teste com o limite inferior:
$$-0.42 < -1.96$$

In [75]:
print(z < z_ha1)

False


<img src="https://drive.google.com/uc?id=1ujfRnPgw_MEQiAsTze5EX6IqBP9gpKW_" width="600">

Com base na avaliação numérica e gráfica, o valor não cai na região de rejeição, portanto **não rejeitamos $H_0$**. Os resultados do teste **não** são estatisticamente significativos ao nível de 5%.

**$$H_{A2}: \mu < 70.8$$**

Lembre-se do valor crítico para $H_{A2}$:
$$z_{A2} = -z_{\alpha} = -1.64$$

O estatística do teste ($z \approx -0.42$) cai na região de rejeição?
$$-0.42 < -1.64$$

In [ ]:
# Rejeita H0?
print(z > z_ha3)

False


<img src="https://drive.google.com/uc?id=1jia44ykabqJpIboIWOzO9lYSD7IdfURx" width="600">

Com base na avaliação numérica e gráfica, o valor não cai na região de rejeição, portanto **não rejeitamos $H_0$**. Os resultados do teste **NÃO** são estatisticamente significativos ao nível de 5%.

**Passo 6: Interpretar o resultado do teste de hipótese.**

**$$H_{A1}: \mu \neq 70.8$$**

Ao nível de significância de 5%, os dados não fornecem evidências suficientes para concluir que o peso médio dos estudantes difere do peso médio dos adultos europeus.

**$$H_{A2}: \mu < 70.8$$**

Ao nível de significância de 5%, os dados não fornecem evidências suficientes para concluir que o peso médio dos estudantes é menor que o peso médio dos adultos europeus.

**$$H_{A3}: \mu > 70.8$$**

Ao nível de significância de 5%, os dados não fornecem evidências suficientes para concluir que o peso médio dos estudantes é maior que o peso médio dos adultos europeus.



### **Teste de hipótese: A abordagem do p-valor**

**Passo 1: Definir a hipótese nula, $H_0$, e a hipótese alternativa, $H_A$.**

A hipótese nula afirma que o peso médio dos estudantes ($\mu$) é igual ao peso médio dos adultos europeus de 70.8 kg ($\mu_0$), conforme relatado por Walpole et al. (2012). Em outras palavras, não há diferença no peso médio dos estudantes e no peso médio dos adultos europeus.
$$H_0: \mu = 70.8$$

Para fins de ilustração, testaremos três hipóteses alternativas:

**Hipótese alternativa 1:** O peso médio dos estudantes **não é igual** ao peso médio dos adultos europeus. Em outras palavras, *existe* uma diferença no peso médio dos estudantes e no peso médio dos adultos europeus.
$$H_{A1}: \mu \neq 70.8$$

**Hipótese alternativa 2:** O peso médio dos estudantes é **menor que** o peso médio dos adultos europeus.
$$H_{A2}: \mu < 70.8$$

**Hipótese alternativa 3:** O peso médio dos estudantes é **maior que** o peso médio dos adultos europeus.
$$H_{A3}: \mu > 70.8$$

**Passo 2: Decidir o nível de significância, $\alpha$.**
$$\alpha = 0.05$$

In [ ]:
alpha = 0.05

**Passo 3: Calcular o valor do estatística do teste.**

A seguinte equação é aplicada para calcular o estatística do teste $z$:

$$z = \frac{\bar{x} - \mu_0}{\sigma / \sqrt{n}}$$

Aplicado ao nosso exemplo, calculamos $z$ por:

$$z = \frac{\bar{x} - \mu_0}{\sigma / \sqrt{n}} = \frac{69.82 - 70.8}{8.64 / \sqrt{14}} \approx -0.42$$

In [ ]:
z = (mean_sample - mu_0) / (sigma / np.sqrt(n))

print(z)

-0.4240454748453689


**Passo 4B: Determinar o p-valor.**

Para calcular o p-valor com Python, aplicamos a função `norm.cdf()` derivada do pacote `scipy` para calcular a probabilidade de ocorrência do estatística do teste com base na **distribuição normal padrão**.

Lembre-se, estamos testando três hipóteses alternativas ($H_{A1}$, $H_{A2}$ e $H_{A3}$). Portanto, calculamos três p-valores também ($P(z_{A1})$, $P(z_{A2})$ e $P(z_{A3})$).

**$H_{A1}: \mu \neq 70.8$**

In [ ]:
# a probabilidade de observar qualquer valor-z maior ou menor em magnitude, dado que a hipótese nula é verdadeira
lower = stats.norm.cdf(z)
upper = 1 - stats.norm.cdf(np.abs(z))
upper

p_value_1 = lower + upper
p_value_1

np.float64(0.6715326491107145)

In [ ]:
lower

np.float64(0.3357663245553573)

In [ ]:
z

np.float64(-0.4240454748453689)

<img src="https://drive.google.com/uc?id=1ZLD_HmFDNL2YuUTJKIoTHBl_1d7smuVf" width="600">

In [ ]:
# a probabilidade de observar um valor-z menor em magnitude dado que a hipótese nula é verdadeira
p_value_2 = stats.norm.cdf(z)
p_value_2

np.float64(0.3357663245553573)

Do Passo 3, o valor do estatística do teste é $z \approx -0.42$. O teste é **unicaudal à esquerda** (left-tailed), então o p-valor é a probabilidade de observar um valor $z$ de -0.42 ou menor em magnitude. Essa probabilidade corresponde à área colorida na figura abaixo, que é 0.336. Assim, $p \approx 0.336$.

<img src="https://drive.google.com/uc?id=1Paa5C_uwWeTGX8VTSjFaPaG4DUqqP1Ob" width="600">

In [ ]:
p_value_3 = 1 - stats.norm.cdf(z)
p_value_3

np.float64(0.6642336754446427)

Do Passo 3, o valor do estatística do teste é $z \approx -0.42$. O teste é **unicaudal à direita** (right-tailed), então o p-valor é a probabilidade de observar um valor $z$ de -0.42 ou maior em magnitude. Essa probabilidade corresponde à área colorida na figura abaixo, que é 0.664. Assim, $p \approx 0.664$.

<img src="https://drive.google.com/uc?id=1PplBEGVerGjExI04frm1V1uJM28EFZL0" width="600">

**Passo 5B: Se $p \le \alpha$, rejeite $H_0$; caso contrário, não rejeite $H_0$**

O p-valor do estatística do teste encontrado no Passo 4 é comparado ao nível de significância $\alpha$ de 5% definido pelo usuário. Lembre-se, estamos investigando três hipóteses alternativas ($H_{A1}$, $H_{A2}$ e $H_{A3}$). Assim, fazemos comparações para cada hipótese em particular.

**$H_{A1}: \mu \neq 70.8$**

$$0.672 \le 0.05$$

In [ ]:
# rejeita H0?
p_value_1 <= alpha

np.False_

**Passo 5B: Se $p \le \alpha$, rejeite $H_0$; caso contrário, não rejeite $H_0$**

O p-valor do estatística do teste encontrado no Passo 4 é comparado ao nível de significância $\alpha$ de 5% definido pelo usuário. Lembre-se, estamos investigando três hipóteses alternativas ($H_{A1}$, $H_{A2}$ e $H_{A3}$). Assim, fazemos comparações para cada hipótese em particular.

**$H_{A1}: \mu \neq 70.8$**

$$0.672 \le 0.05$$

In [76]:
# rejeita H0?
p_value_2 <= alpha

np.False_

O p-valor é maior que o nível de significância especificado de 0.05. Nós **não rejeitamos $H_0$**. Os resultados do teste **NÃO** são estatisticamente significativos ao nível de 5% e não fornecem evidências suficientes contra a hipótese nula.

**$$H_{A2}: \mu < 70.8$$**

$$0.336 \le 0.05$$

In [ ]:
# rejeita H0?
p_value_3 <= alpha

np.False_

O p-valor é maior que o nível de significância especificado de 0.05. Nós **não rejeitamos $H_0$**. Os resultados do teste **NÃO** são estatisticamente significativos ao nível de 5% e não fornecem evidências suficientes contra a hipótese nula.

**Passo 6: Interpretar o resultado do teste de hipótese.**

**$H_{A1}: \mu \neq 70.8$**

Ao nível de significância de 5%, os dados não fornecem evidências suficientes para concluir que o peso médio dos estudantes difere do peso médio dos adultos europeus.

**$H_{A2}: \mu < 70.8$**

Ao nível de significância de 5%, os dados não fornecem evidências suficientes para concluir que o peso médio dos estudantes é menor que o peso médio dos adultos europeus.

**$H_{A3}: \mu > 70.8$**

Ao nível de significância de 5%, os dados não fornecem evidências suficientes para concluir que o peso médio dos estudantes é maior que o peso médio dos adultos europeus.

In [ ]:
p_value_3

np.float64(0.6642336754446427)

**Passo 6: Interpretar o resultado do teste de hipótese.**

**$H_{A1}: \mu \neq 70.8$**
$p \approx 0.672$. Ao nível de significância de 5%, os dados não fornecem evidências suficientes para concluir que o peso médio dos estudantes difere do peso médio dos adultos europeus.

**$H_{A2}: \mu < 70.8$**
$p \approx 0.336$. Ao nível de significância de 5%, os dados não fornecem evidências suficientes para concluir que o peso médio dos estudantes é menor que o peso médio dos adultos europeus.

**$H_{A3}: \mu > 70.8$**
$p \approx 0.664$. Ao nível de significância de 5%, os dados não fornecem evidências suficientes para concluir que o peso médio dos estudantes é maior que o peso médio dos adultos europeus.

## **Testes de Hipótese para uma Média Populacional quando Sigma é Desconhecido**

Um teste de hipótese para uma média populacional quando o desvio padrão populacional, $\sigma$, é desconhecido, é conduzido da mesma forma como se o desvio padrão populacional fosse conhecido. A única diferença é que a **distribuição t** é invocada em vez da distribuição normal padrão (distribuição z).

Para um teste com a hipótese nula $H_0: \mu = \mu_0$, o **estatística do teste, $t$**, é calculada como:

$$t = \frac{\bar{x} - \mu_0}{s / \sqrt{n}}$$

Este procedimento de teste de hipótese é chamado de **teste t para uma média** ou simplesmente **teste t**. Lembre-se que os testes de hipótese seguem um procedimento passo a passo, que é resumido a seguir:

1.  Definir a hipótese nula $H_0$ e a hipótese alternativa $H_A$.
2.  Decidir o nível de significância, $\alpha$.
3.  Calcular o valor do estatística do teste.
4.  **Abordagem do valor crítico:** Determinar o valor crítico.
5.  **Abordagem do p-valor:** Determinar o p-valor.
6.  **Abordagem do valor crítico:** Se o valor da estatística do teste cair na **região de rejeição**, rejeite $H_0$; caso contrário, **não rejeite $H_0$**.
7.  **Abordagem do p-valor:** Se $p \le \alpha$, rejeite $H_0$; caso contrário, **não rejeite $H_0$**.
8.  Interpretar o resultado do teste de hipótese.

Similar à seção anterior, apresentaremos a **abordagem do valor crítico** primeiro e, então, em um segundo passo, repetiremos a análise usando a **abordagem do valor-p**. No entanto, desta vez, vamos encapsular a abordagem do valor crítico dentro de uma função construída por nós. Para a abordagem do valor-p, usaremos o Python e aplicaremos a função `stats.ttest_1samp()` do pacote `scipy`.

### **Teste de hipótese: A abordagem do valor crítico**

Vamos implementar a **abordagem do valor crítico** construindo uma função chamada `simple_t_test()`. A função recebe como argumentos de entrada:

* os dados da amostra (um objeto `Series` do `pandas` chamado `data`).
* a média populacional contra a qual testar ($\mu_0$ (mu0)).
* o nível de significância especificado ($\alpha$ (alpha)).
* a abordagem metodológica do teste t (`left` (esquerda), `right` (direita) ou `two_sided` (bicaudal) como valor padrão).

A saída da função é um valor booleano, que indica se a hipótese nula $H_0$ deve ser rejeitada com base no resultado do teste. Se `VERDADEIRO` (TRUE), $H_0$ é rejeitada, se `FALSO` (FALSE), $H_0$ não é rejeitada.

In [ ]:
from scipy.stats import t

def simple_t_test(data, mu0, alpha, method = "two_sided"):
    """
    Executa um teste t de uma amostra (one-mean t-test) usando a
    abordagem do valor crítico.

    Parâmetros:
    ----------
    data : pandas.Series ou array-like.
        Os dados da amostra.
    mu0 : float
        O valor da média populacional para testar contra (Hipótese Nula H0).
    alpha : float
        O nível de significância (ex: 0.05 para 5%).
    method : str, opcional
        O tipo de teste a ser realizado:
        - "two_sided" (bicaudal, padrão): H_A: mu != mu0
        - "left" (unicaudal à esquerda): H_A: mu < mu0
        - "right" (unicaudal à direita): H_A: mu > mu0

    Retorna:
    -------
    reject : bool
        True se a Hipótese Nula (H0) é rejeitada, False caso contrário.
    """

    # --- Passo 1: Calcular as estatísticas da amostra ---

    # Calcula a média da nossa amostra (x-barra)
    sample_mean = np.mean(data)

    # Calcula o desvio padrão da amostra (s)
    # Nota: np.std() por padrão usa o divisor 'n' (populacional).
    # Para o teste t, o correto é usar 'n-1' (amostral), então usamos ddof=1.
    # (ddof = Delta Degrees of Freedom)
    sample_std = np.std(data, ddof=1)

    # Tamanho da amostra (n)
    n = data.size

    # --- Passo 2: Calcular a estatística do teste (t-score) ---

    # Esta é a fórmula do t-score: (média_amostral - média_H0) / erro_padrão
    # O erro padrão é: desvio_padrão_amostral / raiz_quadrada(n)
    standard_error = sample_std / np.sqrt(n)
    empirical_t = (sample_mean - mu0) / standard_error

    # --- Passo 3: Determinar o valor crítico e tomar a decisão ---

    # Graus de liberdade (df) = n - 1
    # A distribuição t depende do tamanho da amostra (via graus de liberdade)
    df = n - 1

    # Inicializa a variável de decisão
    reject = False

    # --- Lógica para o Teste Unicaudal à Esquerda (H_A: mu < mu0) ---
    if (method == "left"):
        # Encontra o valor crítico t na cauda esquerda da distribuição.
        # t.ppf() (Percent Point Function) é o inverso da cdf.
        # Pedimos o valor t que deixa 'alpha' (ex: 5%) de probabilidade à sua esquerda.
        critical_value = t.ppf(alpha, df = df)

        # Decisão: Rejeitamos H0 se nosso estatística do teste (empirical_t)
        # for *mais extrema* (mais à esquerda) que o valor crítico.
        if (empirical_t < critical_value):
            reject = True

    # --- Lógica para o Teste Unicaudal à Direita (H_A: mu > mu0) ---
    elif (method == "right"):
        # Encontra o valor crítico t na cauda direita.
        # Como t.ppf() olha da esquerda para a direita, pedimos o ponto
        # que deixa '1 - alpha' (ex: 95%) de probabilidade à sua esquerda.
        critical_value = t.ppf(1 - alpha, df = df)

        # Decisão: Rejeitamos H0 se nossa estatística do teste
        # for *mais extremo* (mais à direita) que o valor crítico.
        if (empirical_t > critical_value):
            reject = True

    # --- Lógica para o Teste Bicaudal (H_A: mu != mu0) ---
    else: # "two_sided"
        # O nível de significância 'alpha' (ex: 5%) é dividido em duas caudas.
        # Procuramos os valores críticos que deixam 'alpha / 2' (ex: 2.5%) em cada cauda.
        # Por simetria, basta encontrar o valor da cauda esquerda.
        # O valor da cauda direita será o mesmo, mas positivo.
        critical_value = t.ppf(alpha / 2, df = df)

        # Decisão: Rejeitamos H0 se nossa estatística do teste (em valor absoluto)
        # for maior que o valor crítico (em valor absoluto).
        # Isso cobre os casos de t ser muito positivo ou muito negativo.
        # A lógica original `(-np.abs(empirical_t) < -np.abs(critical_value))` é
        # equivalente a `np.abs(empirical_t) > np.abs(critical_value)`.
        if (np.abs(empirical_t) > np.abs(critical_value)):
            reject = True

        # O valor crítico para o print bicaudal é geralmente mostrado como positivo
        critical_value = np.abs(critical_value)

    # --- Passo 4: Exibir os resultados ---
    print("--- Resultados do Teste t (Abordagem do Valor Crítico) ---")
    print(f"Hipótese Nula (H0): mu = {mu0}")
    print(f"Tipo de Teste: {method}")
    print(f"Nível de significância (alpha): {alpha}")
    print(f"Graus de liberdade (df): {df}")
    print("---------------------------------------------------------")
    print(f"estatística do teste (t-score): {empirical_t:.4f}")

    if (method == "two_sided"):
        print(f"Valores Críticos (±): ±{critical_value:.4f}")
    else:
        print(f"Valor Crítico: {critical_value:.4f}")

    print(f"Rejeitar H0: {reject}")
    print("---------------------------------------------------------")

    return reject

### **Teste t para uma média: Um exemplo**

Agora é hora de testar nossa função `simple_t_test()`. Para isso, refazemos o exemplo da seção anterior. Usamos o conjunto de dados `students`. Você pode baixar o arquivo `students.csv` [aqui](https://www.google.com/search?q=PLACEHOLDER_LINK) e importá-lo do seu sistema de arquivos local, ou carregá-lo diretamente como um recurso da web. Em ambos os casos, você importa o conjunto de dados para o Python como um objeto DataFrame do `pandas` usando o método `read_csv`:

In [ ]:
students.head()

,stud.id,name,gender,age,height,weight,religion,nc.score,semester,major,minor,score1,score2,online.tutorial,graduated,salary
1,833917,"Gonzales, Christina",Female,19,160,64.8,Muslim,1.91,1st,Political Science,Social Sciences,NaN,NaN,0,0,NaN
2,898539,"Lozano, T'Hani",Female,19,172,73.0,Other,1.56,2nd,Social Sciences,Mathematics and Statistics,NaN,NaN,0,0,NaN
3,379678,"Williams, Hanh",Female,22,168,70.6,Protestant,1.24,3rd,Social Sciences,Mathematics and Statistics,45.0,46.0,0,0,NaN
4,807564,"Nem, Denzel",Male,19,183,79.7,Other,1.37,2nd,Environmental Sciences,Mathematics and Statistics,NaN,NaN,0,0,NaN
5,383291,"Powell, Heather",Female,21,175,71.4,Catholic,1.46,1st,Environmental Sciences,Mathematics and Statistics,NaN,NaN,0,0,NaN


O conjunto de dados `students` consiste em 8239 linhas, cada uma representando um estudante específico, e 16 colunas, cada uma correspondendo a uma variável/feature (característica) relacionada àquele estudante. Estas variáveis, cujos nomes são autoexplicativos, são:

  * `stud.id`
  * `name`
  * `gender`
  * `age`
  * `height`
  * `weight`
  * `religion`
  * `nc.score`
  * `semester`
  * `major`
  * `minor`
  * `score1`
  * `score2`
  * `online.tutorial`
  * `graduated`
  * `salary`

Examinamos o `weight` (peso) médio dos estudantes de uma amostra aleatória do conjunto de dados `students` e o comparamos com o `weight` (peso) médio de adultos europeus. Walpole et al. (2012) publicaram dados sobre a massa corporal média (kg) por região, incluindo a Europa. Eles relatam que a massa corporal média da população adulta europeia é de **70.8 kg**. Nós definimos a média populacional adequadamente: $\mu_0 = 70.8$.

Além disso, pegamos uma amostra aleatória (`sample_weights`) com tamanho $n = 9$. A amostra consiste nos `weights` (pesos) em `kg` de 9 estudantes escolhidos aleatoriamente do conjunto de dados `students`.

In [77]:
mu_0 = 70.8
n = 9

sample_weights = students.sample(n, random_state = 9)["weight"]

### **Teste de hipótese**

**Passo 1: Definir a hipótese nula, $H_0$, e a hipótese alternativa, $H_A$.**

A hipótese nula afirma que o peso médio dos estudantes ($\mu$) é igual ao peso médio dos adultos europeus de 70.8 kg ($\mu_0$), conforme relatado por Walpole et al. (2012). Em outras palavras, não há diferença no peso médio dos estudantes e no peso médio dos adultos europeus.
$$H_0: \mu = 70.8$$

Lembre-se, a formulação da hipótese alternativa dita se aplicamos um teste de hipótese bicaudal, unicaudal à esquerda ou unicaudal à direita. Por isso, definimos as três seguintes hipóteses alternativas:

$H_{A1}: \mu \neq 70.8$
resulta em um teste de hipótese **bicaudal**.

$H_{A2}: \mu < 70.8$
resulta em um teste de hipótese **unicaudal à esquerda**.

$H_{A3}: \mu > 70.8$
resulta em um teste de hipótese **unicaudal à direita**.

**Passo 2: Decidir o nível de significância, $\alpha$.**
$$\alpha = 0.05$$

In [ ]:
alpha = 0.05

**Passos 3, 4 e 5: Calcular o valor da estatística do teste, determinar o valor crítico e avaliar o valor da estatística do teste. Se cair na região de rejeição, rejeite $H_0$; caso contrário, não rejeite $H_0$.**

Agora, nossa função construída por nós, `simple_t_test()`, entra em jogo. Nós fornecemos à função uma amostra aleatória na forma de uma `Series` do `pandas`, um valor para $\mu_0$, um nível de significância $\alpha$ e o método (`two_sided`, `left` ou `right`). Lembre-se que `two_sided` é o valor padrão. Assim, se não especificarmos nenhum método, a função aplicará o teste de hipótese bicaudal:

In [ ]:
simple_t_test(sample_weights, mu_0, alpha)

--- Resultados do Teste t (Abordagem do Valor Crítico) ---
Hipótese Nula (H0): mu = 70.8
Tipo de Teste: two_sided
Nível de significância (alpha): 0.05
Graus de liberdade (df): 8
---------------------------------------------------------
estatística do teste (t-score): 1.1620
Valores Críticos (±): ±2.3060
Rejeitar H0: False
---------------------------------------------------------


False

Executando um teste unicaudal à esquerda por:

In [ ]:
simple_t_test(sample_weights, mu_0, alpha, method = "left")

--- Resultados do Teste t (Abordagem do Valor Crítico) ---
Hipótese Nula (H0): mu = 70.8
Tipo de Teste: left
Nível de significância (alpha): 0.05
Graus de liberdade (df): 8
---------------------------------------------------------
estatística do teste (t-score): 1.1620
Valor Crítico: -1.8595
Rejeitar H0: False
---------------------------------------------------------


False

Executando um teste unicaudal à direita por:

In [ ]:
simple_t_test(sample_weights, mu_0, alpha, method = "right")

--- Resultados do Teste t (Abordagem do Valor Crítico) ---
Hipótese Nula (H0): mu = 70.8
Tipo de Teste: right
Nível de significância (alpha): 0.05
Graus de liberdade (df): 8
---------------------------------------------------------
estatística do teste (t-score): 1.1620
Valor Crítico: 1.8595
Rejeitar H0: False
---------------------------------------------------------


False

**Passo 6: Interpretar o resultado do teste de hipótese.**

Suponha que a estatística do teste, e portanto a média da amostra, caia além do valor crítico, ou seja, na região de rejeição. Nesse caso, concluímos que, ao nível de significância de 5%, os dados fornecem evidências suficientes para **rejeitar $H_0$**. Em contraste, se a estatística do teste, e portanto a média da amostra, cair na região de não rejeição, concluímos que os dados **não fornecem evidências para rejeitar $H_0$**.

### **Teste de hipótese em Python: A abordagem do p-valor**

A segunda abordagem é baseada na atribuição de uma probabilidade ao valor da estatística do teste. Se a estatística do teste for muito extrema, dado que a hipótese nula é verdadeira, uma baixa probabilidade será atribuída à estatística do teste. Em contraste, se a estatística do teste não for nem um pouco extrema, a probabilidade atribuída a ela será muito maior. Essa probabilidade é chamada de **p-valor**.

#### **Vídeo: Como calcular p-valores??**

Assista ao vídeo explicativo sobre **Como calcular valores-p** neste [link](https://youtu.be/JQc3yx0-Q9E?si=9bnB_XQKl2pli2e0):

<img src="https://drive.google.com/uc?id=1O53YTjmdnoOpkI1xoKPnyJdAwgaIV-q-" width="400">

Para calcular o p-valor exato para um dado valor numérico, contamos com software. O método alternativo de "procurar na tabela" é de alguma forma tedioso, pois nem todos os valores numéricos da estatística do teste são dados em tal tabela, e isso causa erros de arredondamento.

Felizmente, o Python métodos para realizar testes t através da função `ttest_1samp()` do módulo `stats` do pacote `scipy`. A função nos permite testar uma média populacional quando $\sigma$ é desconhecido de forma direta. Vejamos o uso da função refazendo o mesmo problema acima, mas desta vez usando a função `ttest_1samp()`:

In [ ]:
test_result = stats.ttest_1samp(sample_weights,
                                mu_0,
                                alternative = "two-sided")

test_result

TtestResult(statistic=np.float64(1.1620080498483334), pvalue=np.float64(0.2787240885983498), df=np.int64(8))

A função `ttest_1samp()` retorna um `objeto` que consiste nas seguintes propriedades:

* `<objeto>.statistic` armazena a estatística do teste real e representa o **valor-t** empírico.
* `<objeto>.pvalue` representa o **p-valor** do teste t realizado.
* `<objeto>.df` representa os **graus de liberdade** (degrees of freedom) com base nas observações fornecidas no conjunto de dados de entrada.

Você recupera as propriedades do objeto por:

In [ ]:
test_result.statistic

np.float64(1.1620080498483334)

Alternativamente, você pode recuperar o valor da estatística do teste via indexação:

In [ ]:
test_result[0]

np.float64(1.1620080498483334)

Da mesma forma, o **p-valor** e os **graus de liberdade** (degrees of freedom) são recuperados por:

In [ ]:
print("p-valor =", test_result.pvalue)
print("p-valor =", test_result[1])

print("Graus de liberdade =", test_result.df)

p-valor = 0.2787240885983498
p-valor = 0.2787240885983498
Graus de liberdade = 8


Além disso, o `objeto` retornado permite calcular o intervalo de confiança com base em um nível de confiança fornecido, através da função `<objeto>.confidence_interval()`. Se nenhum argumento for passado, o intervalo de confiança para $\alpha = 5\%$ é retornado:

In [ ]:
test_result.confidence_interval()

ConfidenceInterval(low=np.float64(68.25124075973771), high=np.float64(78.52653701804006))

Fornecendo uma saída mais agradável (ou "mais formatada"):

In [ ]:
print("Com 95 % de confiança, a verdadeira média está dentro do intervalo de",
      round(test_result.confidence_interval()[0], 2), "e",
      round(test_result.confidence_interval()[1], 2), "kg.")

Com 95 % de confiança, a verdadeira média está dentro do intervalo de 68.25 e 78.53 kg.


## **Testes de Hipótese para Duas Médias Populacionais**

Até agora, focamos em testes de hipótese para uma média populacional. No entanto, em muitas aplicações, queremos comparar as médias de duas ou mais populações. Portanto, primeiro temos que distinguir entre amostras de duas populações que são **independentes** e amostras de populações que não são independentes, as quais são chamadas de amostras **pareadas** (ou dependentes). Nas seções seguintes, discutiremos procedimentos inferenciais para comparar as médias de duas populações.

Nós denotaremos os parâmetros e as estatísticas das populações 1 e 2 com subscritos 1 e 2, respectivamente. Assim, $\mu_1$ e $\sigma_1$ são parâmetros populacionais da população 1, e $\mu_2$ e $\sigma_2$, os da população 2. Da mesma forma, $\bar{x}_1$, $s_1$ e $n_1$ são a média amostral, o desvio padrão amostral e o tamanho da amostra da população 1, enquanto $\bar{x}_2$, $s_2$ e $n_2$ correspondem à amostra extraída da população 2.

Para amostras independentes de tamanhos $n_1$ e $n_2$ das populações 1 e 2, a média de todas as diferenças possíveis entre as duas médias amostrais é igual à diferença entre as duas médias populacionais:
$$\mu_{\bar{x}_1 - \bar{x}_2} = \mu_1 - \mu_2$$

Além disso, o desvio padrão de todas as diferenças possíveis entre as duas médias amostrais (conhecido como erro padrão da diferença entre as médias) é igual à raiz quadrada da soma das variâncias populacionais, cada uma dividida pelo tamanho da amostra correspondente:
$$\sigma_{\bar{x}_1 - \bar{x}_2} = \sqrt{\frac{\sigma_1^2}{n_1} + \frac{\sigma_2^2}{n_2}}$$

Uma variável normalmente distribuída ou um tamanho de amostra grande o suficiente (lembre-se do **Teorema do Limite Central**) faz com que a distribuição da diferença das médias amostrais ($\bar{x}_1 - \bar{x}_2$) também seja normalmente distribuída.

Os procedimentos de teste de hipótese para duas médias populacionais são, na verdade, os mesmos que para uma média populacional. Por favor, note que nas seções seguintes, focaremos na **abordagem do p-valor** e não discutiremos mais a abordagem do valor crítico. Portanto, o procedimento de teste de hipótese é ligeiramente revisado. O procedimento passo a passo para testes de hipótese é resumido da seguinte forma:

1.  Definir a hipótese nula $H_0$ e a hipótese alternativa $H_A$.
2.  Decidir o nível de significância, $\alpha$.
3.  Calcular o valor da estatística do teste.
4.  Determinar o p-valor.
5.  Se $p \le \alpha$, rejeite $H_0$; caso contrário, **não rejeite $H_0$**.
6.  Interpretar o resultado do teste de hipótese.

## **Referência**
Rudolph, A., Krois, J., Hartmann, K. (2023): Statistics and Geodata Analysis using Python ([SOGA-Py](https://www.geo.fu-berlin.de/soga-py)). Department of Earth Sciences, Freie Universitaet Berlin.